bubble_sort_by.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+def bubble_sort_by(arr)
+    if block_given?
+        swap = -1
+        while swap != 0
+            swap = 0
+            (0..(arr.length - 2)).each do |i|
+                if yield(arr[i], arr[i + 1]).positive?
+                    arr[i], arr[i + 1] = arr[i + 1], arr[i]
+                    swap += 1
+                end
+            end
+        end
+    else
+        arr = "no block"
```

Comment:
You must raise an error, here.

Reply:
the error says "No block" when there is no block given, tested with: 
bubble_sort_by([])

Reply:
Didnt know how to work with raise exceptions until now, thank you for it, best regards, its fixed